web/pdf_page_view.js

```diff
@@ -564,17 +575,38 @@ class PDFPageView {
       }
     }
 
+    const postponeDrawing = postponeDrawingDelay >= 0;
+
+    if (postponeDrawing) {
+      this.textLayer?.hide();
+    } else {
+      this.textLayer?.show();
+    }
+
     if (this.canvas) {
       if (
+        postponeDrawing ||
         this.useOnlyCssZoom ||
         (this.hasRestrictedScaling && isScalingRestricted)
       ) {
+        if (postponeDrawingDelay > 0 && isFinite(postponeDrawingDelay)) {
```

Comment:
The same comments about better delay-validation, see `pdf_viewer.js` comments, apply here as well.